app/src/main/java/appricottsoftware/clarity/LoginActivity.java

```diff
@@ -3,44 +3,279 @@
+                super.onFailure(statusCode, headers, throwable, errorResponse);
+            }
+
+            @Override
+            public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSONArray errorResponse) {
+                super.onFailure(statusCode, headers, throwable, errorResponse);
+            }
+
+            @Override
+            public void onFailure(int statusCode, Header[] headers, String responseString, Throwable throwable) {
+                super.onFailure(statusCode, headers, responseString, throwable);
+            }
+
+            @Override
+            public void onSuccess(int statusCode, Header[] headers, String responseString) {
+                super.onSuccess(statusCode, headers, responseString);
+            }
+        });
+        return false;
+    }
+
+    private void facebookLogin() {
+        Profile fbProfile = Profile.getCurrentProfile();
+        fbAccessTokenTracker = new AccessTokenTracker() {
+            @Override
+            protected void onCurrentAccessTokenChanged(AccessToken oldAccessToken,
+                                                       AccessToken currentAccessToken) {
+                if (currentAccessToken == null) {
+                    Log.d("FB token", "null");
+                }
+            }
+        };
+
+        if (fbProfile == null) {
+            fbCallbackManager = CallbackManager.Factory.create();
+            btnFacebook = findViewById(R.id.btn_loginFacebook);
+            btnFacebook.setReadPermissions(Arrays.asList(EMAIL));
+            btnFacebook.registerCallback(fbCallbackManager, new FacebookCallback<LoginResult>() {
+                @Override
+                public void onSuccess(LoginResult loginResult) {
+                    // App code
+                    Log.e("Facebook Login", "onSuccess login Facebook");
+                    GraphRequest request = GraphRequest.newMeRequest(
+                            AccessToken.getCurrentAccessToken(),
+                            new GraphRequest.GraphJSONObjectCallback() {
+                                @Override
+                                public void onCompleted(JSONObject object, GraphResponse response) {
+                                    FacebookSdk.setIsDebugEnabled(true);
+                                    FacebookSdk.addLoggingBehavior(LoggingBehavior.INCLUDE_ACCESS_TOKENS);
+
+                                    Log.e("Facebook Login", "AccessToken.getCurrentAccessToken() " + AccessToken.getCurrentAccessToken().toString());
+                                    Profile profile = Profile.getCurrentProfile();
+                                    Log.e("Facebook Login", "Current profile: " + profile);
+                                    if (profile != null) {
+                                        Log.e("Facebook Login", String.format("id = %s; name = %s; lastName = %s; uri = %s",
```

Comment:
Can you remove the Log.e statements in onSuccess? We probably don't want to log user data.

Reply:
Done